src/index.ts

```diff
@@ -44,7 +44,11 @@ for (const profileId of oauthProfileIds) {
             if (databaseClient) {
 
                 await graphqlClient.saveClient(databaseClient);
-                console.log(`OAuth client '${configClient.id}' was succesfully migrated to database storage`);
+                console.log(`OAuth client '${configClient.id}' was successfully migrated to database storage`);
+
+                if (environment.deleteMigratedClients) {
+                    await restconfClient.deleteClient(profileId, configClient.id);
```

Comment:
Would it be more appropriate to remember the client IDs and do a single operation at the end deleting all clients? That would allow easily rolling back the batch deletion.

Reply:
You are right - that works better I think. The main behavior I wanted was to be re-runnable if the migration went wrong half-way through, which still works when doing deletions at the end.